components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -138,12 +138,23 @@ public void handle(List<String> roles, String subject, Map<String, String> attri
                 }
 
                 if (!userClaims.isEmpty()) {
+                    userClaims.remove(FrameworkConstants.PASSWORD);
                     userStoreManager.setUserClaimValues(username, userClaims, null);
                 }
 
-            } else {
+                UserProfileAdmin userProfileAdmin = UserProfileAdmin.getInstance();
 
-                userStoreManager.addUser(username, generatePassword(), addingRoles.toArray(
+                if (StringUtils.isEmpty(userProfileAdmin.getNameAssociatedWith(idp, subjectVal))) {
+                    // Associate User
+                    associateUser(username, userStoreDomain, tenantDomain, subjectVal, idp);
+                }
```

Comment:
Will this work without a tenant flow. AFAIK, UserProfileAdmin identifies tenant from the realm not from the username